### PR TITLE
web: Make the dev proxy work with newer Node.js versions

### LINKED ISSUE #10689: web ui npm start broken
### What did you do?

- npm install
- npm start then the result is below

### What did you expect to see?

npm start work well

### What did you see instead? Under which circumstances?

image

### System information

mac os

### Prometheus version

_No response_

### Prometheus configuration file

_No response_

### Alertmanager version

_No response_

### Alertmanager configuration file

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by fatelei**
fix #10689

### CODE REVIEW

**Comment by juliusv** [comment]
/cc @Nexucis 

Thanks! Hmm, for me both `npm start` and `npm run start` work fine, so I am unable to reproduce your issue. It also seems to work fine on Gitpod.

Btw. we only use npm, so we probably wouldn't want to introduce a `yarn.lock` file. And we probably don't want another lockfile in `web/`.

**Comment by Nexucis** [comment]
I'm not able to reproduce it too. I tried with a fresh clone and did the following command:

```bash
cd web/ui
npm install
npm run build:module
npm start
```

it works with no issue.

Can you provide the `npm` and `NodeJS` version you are using please ?

**Comment by Nexucis** [comment]
ah maybe it comes from NodeJS v18. We are currently using NodeJS v16. Do you mind to try with this version and check if it works on your side ?

**Comment by fatelei** [comment]
> 

I try v16.0.0, web ui start correctly, but v18.1.0 not work as except

**Comment by fatelei** [comment]
and v16.0.0 using this MR change, work well as except

**Comment by Nexucis** [comment]
thanks @fatelei for the verification.
I would tend to be fine with these changes. If it's frustrated people using NodeJS v18 and they have to downgrade because of NodeJS version issue to compile Prometheus, well at least this PR can solve it and it works also with our current configuration.

Can you remove the file `package-lock.json` in the folder `web` ? We don't need it, probably you add it by mistake ?

**Comment by fatelei** [comment]
> thanks @fatelei for the verification. I would tend to be fine with these changes. If it's frustrated people using NodeJS v18 and they have to downgrade because of NodeJS version issue to compile Prometheus, well at least this PR can solve it and it works also with our current configuration.
> 
> Can you remove the file `package-lock.json` in the folder `web` ? We don't need it, probably you add it by mistake ?

ok, i will remove packge-lock.json for my mistake

**Comment by Nexucis** [comment]
mmm I just tested your PR with NodeJS 16 and actually the proxy doesn't work (at least on my side). I have a weird error saying : 
`You need to enable JavaScript to run this app.`

Even when Prometheus is running, it is still showing this issue.

Do you have something similar on your side with NodeJS 16 ?

**Comment by fatelei** [comment]
> mmm I just tested your PR with NodeJS 16 and actually the proxy doesn't work (at least on my side). I have a weird error saying : `You need to enable JavaScript to run this app.`
> 
> Even when Prometheus is running, it is still showing this issue.
> 
> Do you have something similar on your side with NodeJS 16 ?



> mmm I just tested your PR with NodeJS 16 and actually the proxy doesn't work (at least on my side). I have a weird error saying : `You need to enable JavaScript to run this app.`
> 
> Even when Prometheus is running, it is still showing this issue.
> 
> Do you have something similar on your side with NodeJS 16 ?

it worked
image

setupProxy should `This file only supports Node's JavaScript syntax. Be sure to only use supported language features (i.e. no support for Flow, ES Modules, etc).` from https://create-react-app.dev/docs/proxying-api-requests-in-development/

**Comment by Nexucis** [comment]
closing/opening the PR to trigger the CI

**Comment by Nexucis** [comment]
thanks for the changes @fatelei. It works on my side. I'm fine with the changes.
@juliusv over to you to decide if you are ok as well with the changes :).

**Comment by Nexucis** [comment]
Probably the title of the PR should be reviewed too, based on the investigation.
The issue was more related to the NodeJS version than a conflict in the environment variable

**Comment by fatelei** [comment]
> Probably the title of the PR should be reviewed too, based on the investigation. The issue was more related to the NodeJS version than a conflict in the environment variable

i found this: https://github.com/facebook/create-react-app/issues/11762

**Comment by Nexucis** [comment]
I know :) but as you can see, we didn't set the environment variable `HOST`. That's why I said it's probably another issue more related to NodeJS even if the resolution at the end is what you did (a.k.a using a file to setup the proxy).

But don't worry I was more talking to @juliusv for the rephrasing ;). He is definitely better than me for this part.

File: web/ui/react-app/src/setupProxy.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+module.exports = function(app) {
+  app.use(
+    '/api',
+    createProxyMiddleware({
+      target: 'http://localhost:9090',
+      changeOrigin: true,
+    })
+  );
+};
```
Comment (juliusv) [inline web/ui/react-app/src/setupProxy.js:11]: GitHub is warning about the lack of a newline at the end of the file here, could you add that?

**Comment by juliusv** [comment]
:+1: in general besides my last comment. I updated the PR title to hopefully reflect what it does.

File: web/ui/react-app/src/setupProxy.js (hunk 1 above)
Comment (fatelei) [reply web/ui/react-app/src/setupProxy.js:11]: ok

**Comment by Nexucis** [comment]
closing/opening again the PR to trigger the CI.

**Pull request merged**